File: samples/src/main/java/tk/zielony/carbonsamples/component/ComponentViewActivity.java
package tk.zielony.carbonsamples.component;

import android.os.Bundle;

import androidx.annotation.Nullable;

import tk.zielony.carbonsamples.R;
import tk.zielony.carbonsamples.SampleAnnotation;
import tk.zielony.carbonsamples.ThemedActivity;

@SampleAnnotation(layoutId = R.layout.activity_componentview, titleId = R.string.componentViewActivity_title)
public class ComponentViewActivity extends ThemedActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initToolbar();
    }
}

File: samples/src/main/java/tk/zielony/carbonsamples/component/ComponentViewActivity.kt
package tk.zielony.carbonsamples.component

import android.os.Bundle
import kotlinx.android.synthetic.main.activity_componentview.*
import tk.zielony.carbonsamples.R
import tk.zielony.carbonsamples.SampleAnnotation
import tk.zielony.carbonsamples.ThemedActivity
import tk.zielony.carbonsamples.demo.SongItem
import tk.zielony.carbonsamples.demo.SongRow

@SampleAnnotation(layoutId = R.layout.activity_componentview, titleId = R.string.componentViewActivity_title)
class ComponentViewActivity : ThemedActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initToolbar()

        songComponent.getComponent<SongRow>().data = SongItem(3, "Classes in the Jar", 245)
    }
}